src/components/ContentNode/TabNavigator.vue

```diff
@@ -71,6 +71,21 @@ export default {
       currentTitle: this.titles[0],
     };
   },
+  watch: {
+    titles(newVal, oldVal) {
+      if (newVal.length < oldVal.length) {
```

Comment:
To not add nested ifs, you could consider exit if newVal includes current title and continue with the case of a newVal being greater or equal than the oldVal

```js
// if newVal includes current title exit
if (newVal.includes(this.currentTitle)) return;
// if newVal is greater than or equal than the oldVal
// find its new tab
if (newVal.length >= oldVal.length) {
  const newTab = newVal.find(tab => !oldVal.includes(tab));
  this.currentTitle = newTab || this.currentTitle;
  return;
}
// otherwise set first title as current title
const [firstTitle] = newVal;
this.currentTitle = firstTitle;
```

Reply:
Actually thats not how it should work. If a new items is added, it should be highlighted.

Reply:
Okay, so the checking if the newVal is greater or equal than the oldVal should come up first, right?

```js
// if newVal is greater than or equal than the oldVal
// find its new tab
if (newVal.length >= oldVal.length) {
  const newTab = newVal.find(tab => !oldVal.includes(tab));
  this.currentTitle = newTab || this.currentTitle;
  return;
}
// if newVal includes current title exit
if (newVal.includes(this.currentTitle)) return;
// otherwise set first title as current title
const [firstTitle] = newVal;
this.currentTitle = firstTitle;
```